tests/core/go_binary/meaning2.go

```diff
@@ -0,0 +1,12 @@
+package main
+
+import(
+	"fmt"
+)
+
+//go:noescape
```

Comment:
What is this needed for? Calling convention?

Reply:
It declares that the go runtime can rest assured that the implementation of the method doesn't leak pointers which can "escape" the function onto the heap.
https://gcc.gnu.org/onlinedocs/gccgo/Compiler-Directives.html

Reply:
My question is more about why it is needed here. Would the test not work without it?

Reply:
Oh. Huh! How about that. I guess it's not a problem. Maybe the repo's where I've seen this technique did some linting or something in the past that wouldn't work without that. Removed in 3395669